source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -5,4 +5,140 @@ Access Data From a Cursor
+closes if there is an early termination.
+
+.. code-block:: java
+   :copyable: true
+   
+   MongoCursor<Document> cursor = collection.find().cursor();
+
+Usage Patterns
+--------------
+
+A ``MongoCursor`` and ``FindIterable`` allow you to access query results
+one document at a time, abstracting away network and caching logic. 
+
+Functional Iteration
+~~~~~~~~~~~~~~~~~~~~~
+
+Pass a function to the 
+`forEach() <https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html?is-external=true#forEach(java.util.function.Consumer)>`_ 
+method of a ``FindIterable`` to iterate through results in a functional style:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Cursor.java
+   :language: java
+   :dedent:
+   :start-after: begin forEachIteration
+   :end-before: end forEachIteration
+
+.. warning::
+
+   Initiating methods implement the Iterable interface which allows you
+   to iterate using iterator methods. Sometimes, we use an enhanced
+   for-each loop to iterate through results: 
+
+   .. code-block:: java
+
+      for (Document cur : collection.find()) {
+         ...
+      }
+
+   Iterating this way is not preferable because if an exception is
+   thrown before the loop completes, the cursor will not close. Using a
+   ``MongoCursor`` allows us to ensure the cursor closes as shown in the
+   :ref:`cursor cleanup section <cursor_cleanup>`.
+
+Conditional Iteration
+~~~~~~~~~~~~~~~~~~~~~
+
+Use the :java-docs:`hasNext() <apidocs/mongodb-driver-sync/com/mongodb/client/MongoCursor.html#hasNext()>`
+method to check if there are any documents available in the cursor, and then use the
+:java-docs:`next() <apidocs/mongodb-driver-sync/com/mongodb/client/MongoCursor.html#next()>`
+method to retrieve the next available document from the cursor:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Cursor.java
+   :language: java
+   :dedent:
+   :start-after: begin manualIteration
+   :end-before: end manualIteration
+
+.. _cursor_cleanup:
+
+Cursor Cleanup
+--------------
+
+Close
+~~~~~
+
+Use the
+:java-docs:`close() <apidocs/mongodb-driver-sync/com/mongodb/client/MongoCursor.html#close()>`
+method to free up a cursor's consumption of resources in both the client
+application and the MongoDB server:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Cursor.java
+   :language: java
+   :dedent:
+   :start-after: begin explicitCloseExample
+   :end-before: end explicitCloseExample
+   
+Try with Resources Statement
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use a `try-with-resources statement
+<https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html>`_
+to automatically free up a cursor's consumption of resources in both the
+client application and the MongoDB server:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Cursor.java
+   :language: java
+   :dedent:
+   :start-after: begin implicitCloseExample
```

Comment:
The example included here doesn't use try-with-resources statement anymore.  It looks like you deleted that example, the one that looked like:

   try(MongoCursor<Document> cursor = collection.find().cursor()) {
        // iterate
   }